crates/index-scheduler/src/queue/batches.rs

```diff
@@ -239,33 +239,21 @@ impl BatchQueue {
                 remove_task_datetime(wtxn, self.enqueued_at, enqueued_at.earliest, old_batch.uid)?;
                 remove_task_datetime(wtxn, self.enqueued_at, enqueued_at.oldest, old_batch.uid)?;
             } else {
-                let started_at = old_batch.started_at.unix_timestamp_nanos();
-
-                // We have either one or two enqueued at to remove
-                let mut exit = old_batch.stats.total_nb_tasks.clamp(0, 2);
-                let mut iterator = self.enqueued_at.rev_iter_mut(wtxn)?;
-                while let Some(entry) = iterator.next() {
-                    let (key, mut value) = entry?;
-                    if key > started_at {
-                        continue;
-                    }
-                    if value.remove(old_batch.uid) {
-                        exit = exit.saturating_sub(1);
-                        // Safe because the key and value are owned
-                        unsafe {
-                            iterator.put_current(&key, &value)?;
-                        }
-                        if exit == 0 {
-                            break;
-                        }
-                    }
-                }
+                // If we don't have the enqueued at in the batch it means the database comes from the v1.12
+                // and we still need to find the date by scrolling the database
+                remove_n_tasks_datetime_earlier_than(
+                    wtxn,
+                    self.enqueued_at,
+                    old_batch.started_at,
+                    if old_batch.stats.total_nb_tasks >= 2 { 2 } else { 1 },
```

Comment:
Can we use a clamp here?

```rust
old_batch.stats.total_nb_tasks.clamp(1, 2),
```

Reply:
Done, for consistency I also replaced your if when removing batches with my clamp :grin: 
https://github.com/meilisearch/meilisearch/pull/5272/commits/9f21abd4f300ed4aee2988f8ea09ec799611b878